### ISSUE TITLE
Streaming connections should have auto-reopen option

### ISSUE THREAD

**Issue by lukehutch** [serverpod/serverpod#1683]
Right now you can auto-close the streaming connection by providing a connectivity monitor when the client is created. It closes the connection when connectivity is lost.

However, there is no way to auto-reopen the websocket if you try to send something to it, when the streaming connection is closed. `_sendRawWebSocketMessage` just throws an exception if the socket is not open:

```dart
  Future<void> _sendRawWebSocketMessage(String message) async {
    if (_webSocket == null) {
      throw const ServerpodClientException('WebSocket is not connected', 0);
    }
    _webSocket!.sink.add(message);
  }
```

This means that if there is any chance a user's network connection can drop out, suddenly streaming calls will start to fail once the user gets back online.

It is possible to set up a listener to the connectivity monitor to reopen the streaming connection when the client is back online, but it seems like that should be done automatically.

Basically I wonder if `_sendRawWebSocketMessage` should call `openStreamingConnection` before every attempt to send a message, so that the websocket is reopened whenever it is closed.

**Comment by vlidholt** [comment]
Yes, there is already a class for handling reconnecting automatically `StreamingConnectionHandler`. Check Pixorama for an example of how to use it: https://github.com/serverpod/pixorama

**Comment by lukehutch** [comment]
Thanks, although there are problems with it: #1687
